File: problem2.fs
List.append [1] [2]

let rec fibo limit a:int list =
    match limit with
    |0 -> a
    |_ -> if a.[a.Length-1] > limit then a else List.append a [a.[a.Length-2] + a.[a.Length-1]] |> fibo (limit-1) 

fibo 50 [1;2]

// no 

File: problem2.js
var result = 0;

function Fibo(a, b, limit){
    // console|.log(a+b);
    if((a+b) % 2 === 0)
        result += a+b;
    if(b + a+b < limit)
        Fibo(b, a+b, limit);
}

Fibo(0,1,4000000);
console.log(result)